### PR TITLE
Ignore args whose name start with an underscore.

### PR DESCRIPTION
**Pull request by anntzer**
See #34.

### CODE REVIEW

File: test_defopt.py [hunk 1]
```diff
@@ -674,6 +674,11 @@ def foo(*bar):
             return bar
         self.assertNotIn('default', self._get_help(foo))
 
+    def test_private(self):
+        def foo(bar, _baz=None):
+            """:param int bar: bar help"""
+            self.assertNotIn('baz', self._get_help(foo))
```
Comment (evanunderscore) [inline test_defopt.py]: This is indented one level too deep and won't be run.

File: defopt.py [hunk 2]
```diff
@@ -158,7 +158,10 @@ class _NoTypeFormatter(_Formatter):
 
 
 def _populate_parser(func, parser, parsers, short):
-    sig = _inspect_signature(func)
+    full_sig = _inspect_signature(func)
+    sig = full_sig.replace(
+        parameters=[param for param in full_sig.parameters.values()
+                    if not param.name.startswith('_')])
```
Comment (evanunderscore) [inline defopt.py]: Can I get you to use `list(...)` here for the same reason you have `dict` and `set` below?

**Comment by evanunderscore** [comment]
Thanks for the fix. Will merge after the requested minor changes.

**Comment by anntzer** [comment]
fixed

**Pull request merged**